opendemic/estimation/rt/model.py

```diff
@@ -0,0 +1,274 @@
+	all_estimates = []
+	for fips in all_fips:
+		try:
+			county_data = counties_full.xs(fips)
+		except KeyError as ke:
+			logger.error(ke)
+		if len(county_data) <= 1:
+			continue
+		county_name = county_data.iloc[0]['county']
+		state_name = county_data.iloc[0]['state']
+		country_code = 'USA'
+		cases = county_data['cases']
+		_, smoothed, index_dates = prepare_cases(cases=cases)
+		try:
+			rt, low, high = compute_rt(smoothed_new_cases=smoothed)
+		except TimeSeriesDimensionError	as e:
+			logger.error(e)
+			continue
+		rt_region_estimate = DataFrame({
+			'rt_estimate': rt,
+			'rt_low': low,
+			'rt_high': high,
+			'date': index_dates,
+			'country_code': country_code,
+			'first_administrative_division_name': state_name,
+			'region_code': fips,
+			'region_name': county_name
+		})
+		all_estimates.append(rt_region_estimate)
+
+	all_estimates_df = pd.concat(all_estimates)
+	all_estimates_df.reset_index(inplace=True, drop=True)
+	database_connection = sqlalchemy.create_engine(generate_db_uri())
+	all_estimates_tbl_name: str = 'rt_estimates'
+	all_estimates_tmp_tbl_name: str = 'rt_estimates_tmp'
+	all_estimates_df.to_sql(name=all_estimates_tmp_tbl_name, con=database_connection, if_exists='replace', index=False)
+	with database_connection.begin() as cnx:
+		insert_sql: str = """
+			INSERT IGNORE INTO {}({}, `modified`)
+			SELECT {}, UTC_TIMESTAMP() FROM {} as t
+			ON DUPLICATE KEY UPDATE
+				`rt_estimate` = t.`rt_estimate`,
+				`rt_low` = t.`rt_low`,
+				`rt_high` = t.`rt_high`,
+				`modified` = UTC_TIMESTAMP()
+		""".format(
+			all_estimates_tbl_name,
+			', '.join(list(all_estimates_df.columns)),
+			', '.join(list(all_estimates_df.columns)),
+			all_estimates_tmp_tbl_name
+		)
+		cnx.execute(insert_sql)
+		delete_tmp_table_sql: str = "DROP TABLE IF EXISTS {}".format(all_estimates_tmp_tbl_name)
+		cnx.execute(delete_tmp_table_sql)
+
+
+def fetch_rt_estimate(fips: str) -> Tuple[date, List[date], List[float], List[float], List[float]]:
+	rdb: RDBManager = RDBManager(reader=True)
+	last_updated: date = None
+	_date: List[date] = None
+	rt_estimate: List[float] = None
+	rt_low: List[float] = None
+	rt_high: List[float] = None
+	rt_estimates, err = rdb.execute(
+		sql_query="""
+			SELECT
+				`date`,
+				`rt_estimate`,
+				`rt_low`,
+				`rt_high`,
+				DATE(MAX(`modified`)) AS `last_updated`
+			FROM `rt_estimates`
+			WHERE
+				`country_code` = 'USA'
+				AND
+				`region_code` = '{}'
+			GROUP BY 
+				`date`,
+				`rt_estimate`,
+				`rt_low`,
+				`rt_high`
+			ORDER BY `date` ASC
+		""".format(fips)
+	)
+	if err is not None:
+		logger.error(err)
+	if len(rt_estimates) > 0:
+		last_updated = rt_estimates[0]['last_updated']
+		_date = [item['date'] for item in rt_estimates]
+		rt_estimate = [float(item['rt_estimate']) for item in rt_estimates]
+		rt_low = [float(item['rt_low']) for item in rt_estimates]
+		rt_high = [float(item['rt_high']) for item in rt_estimates]
+	return last_updated, _date, rt_estimate, rt_low, rt_high
```

Comment:
I'd suggest creating an object, that contains all these fields. Something like RTestimate. This way also the signature of the function will be much more readable: fetch_rt_estimate(fips: str) -> RTestimate